src/app/todo/components/create/create.component.ts

```diff
@@ -1,17 +1,27 @@
 import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { TodoService } from '../../services/todo.service';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-create',
   templateUrl: './create.component.html',
   styleUrls: ['./create.component.scss'],
 })
 export class CreateComponent implements OnInit {
+  createForm = new FormGroup({
+    name: new FormControl(null, Validators.required),
+    time: new FormControl(null, Validators.required),
+  });
 
-
-  constructor() {
+  constructor(private  todoService: TodoService, private router: Router) {
   }
 
   ngOnInit(): void {
   }
 
+  post(): void {
+    this.todoService.createTask(this.createForm.value);
+    this.router.navigate(['/home']);
```

Comment:
this can create a problem
because the navigation will happen at the same time as the posting of the new task 
so when the user return to the homepage, he may find that the task hasn't been added

please navigate the user back only after the http request is done